redis/connection.py

```diff
@@ -381,20 +378,21 @@ def _read_response(self, disable_decoding=False):
         elif byte == b":":
             response = int(response)
         # bulk response
-        elif byte == b"$":
-            length = int(response)
-            if length == -1:
-                return None
-            response = self._buffer.read(length)
+        elif byte == b"$" and response == b"-1":
+            return None
+        elif byte == b"$" and response != b"-1":
+            response = self._buffer.read(int(response))
         # multi-bulk response
-        elif byte == b"*":
-            length = int(response)
-            if length == -1:
-                return None
+        elif byte == b"*" and response == b"-1":
+            return None
+        elif byte == b"*" and response != b"-1":
             response = [
                 self._read_response(disable_decoding=disable_decoding)
-                for i in range(length)
+                for i in range(int(response))
             ]
+        elif byte not in (b"-", b"+", b":", b"$", b"*"):
```

Comment:
Maybe this line could be just `elif byte != b"+":`. All other comparisons are known to be true here because none of the previous ifs/elifs matched.

Reply:
@Ernest0x please take another look.. I've actually made it simpler.. and everything passes currently. But - I would love more eyes.

Reply:
@chayim looks good. A minor thing could be that `int(response)` which is done in several places. Perhaps a variable could be set for that.

Reply:
The problem is that doing that adds overhead and slows the parsing down (a bit, sure).  This is one of those cases where the speed improvement dictates something that is less "PEP 1"... But then again, it follows the [zen](https://peps.python.org/pep-0020/)